src/imgui/ImGuiSymbols.cc

```diff
@@ -189,15 +267,60 @@ void ImGuiSymbols::paint(MSXMotherBoard* /*motherBoard*/)
 
 		im::TreeNode("Symbols per file", ImGuiTreeNodeFlags_DefaultOpen, [&]{
 			auto drawFile = [&](const FileInfo& info) {
+				auto* file = symbolManager.findFile(info.filename);
 				im::StyleColor(!info.error.empty(), ImGuiCol_Text, getColor(imColor::ERROR), [&]{
 					auto title = strCat("File: ", info.filename);
 					im::TreeNode(title.c_str(), [&]{
 						if (!info.error.empty()) {
 							ImGui::TextUnformatted(info.error);
 						}
 						im::StyleColor(ImGuiCol_Text, getColor(imColor::TEXT), [&]{
+							auto arrowSize = ImGui::GetFrameHeight();
+							auto extra = arrowSize + 2.0f * style.FramePadding.x;
+							ImGui::SetNextItemWidth(ImGui::CalcTextSize("3-3").x + extra);
+						        ImGui::AlignTextToFramePadding();
+
+							// build slot-subslot preview string
+							std::string preview = file->slot ? (isSlotExpanded(motherBoard, *file->slot & 3) ? tmpStrCat(*file->slot & 3, '-', (*file->slot >> 2) & 3).c_str() : tmpStrCat(*file->slot & 3).c_str()) : tmpStrCat("-").c_str();
```

Comment:
This is a very long line, not easy to see what's going on. It also contains some weird stuff:
* You repeat the same subexpressions multiple times. Local variables are useful ;-)
* `tmpStrCat("-").c_str()` is just `"-"`, no?
* You want the result as `std::string`, then you should use `strCat()` not `tmpStrCat()` (this saves a conversion from `TemporaryString` to `const char*` (via `.c_str()`) to `std::string`.
```
auto preview = [&]{
    if (file->slot) {
        int ps = *file->slot & 3;
        int ss = (*file->slot >> 2) & 3;
        return isSlotExpanded(motherBoard, ps) ? strCat(ps, '-', ss)
                                               : strCat(ps)
    } else {
        return std::string("-");
    }
}();
```
Sure that's a bit longer than 1 line, but IMHO much clearer.